packages/transpiler/src/index.ts

```diff
@@ -0,0 +1,114 @@
+import { getContract, isContract, throwIfInvalidNode } from './solc/ast-utils';
+import { transpile } from './transpiler';
+import {
+  transformConstructor,
+  transformContractName,
+  appendDirective,
+  prependBaseClass,
+  purgeContracts,
+  transformParents,
+  fixImportDirectives,
+  purgeVarInits,
+} from './transformations/index';
+import { getInheritanceChain } from './solc/get-inheritance-chain';
+import { Artifact } from './solc/artifact';
+import { Transformation } from './transformation';
+
+export interface OutputFile {
+  fileName: string;
+  source: string;
+  path: string;
+  contracts: string[];
+}
+
+interface FileTran {
+  transformations: Transformation[];
+  source: string;
+}
+
+export function transpileContracts(contracts: string[], artifacts: Artifact[]): OutputFile[] {
+  // check that we have valid ast tree
+  for (const art of artifacts) {
+    throwIfInvalidNode(art.ast);
+  }
+
+  // create contract name | id to artifact map for quick access to artifacts
+  const contractsToArtifactsMap = artifacts.reduce<Record<string | number, Artifact>>((acc, art) => {
```

Comment:
>I don't think this is correct if there's more than one contract with the same name in a project.

`oz compile` doesn't support having two contracts with the same name in a project right now. It will produce a corrupt build files if you do that.

How do we expect to handle contracts with the same in a project? What would be their artifacts called? Because right now contract `Counter` has artifact called `Counter.json`.